tests/reporting/test_market_unit.py

```diff
@@ -0,0 +1,527 @@
+from datetime import timedelta
+from utils import longToHexString, stringToBytes, bytesToHexString
+from pytest import fixture, raises
+from ethereum.tools.tester import ABIContract, TransactionFailed
+
+
+def test_market_creation(localFixture, mockUniverse, mockReportingWindow, mockCash, chain, constants, mockMarket, mockReputationToken, mockShareToken, mockShareTokenFactory):
+    numTicks = 10 ** 10
+    fee = 1
+    oneEther = 10 ** 18
+    endTime = chain.head_state.timestamp + constants.DESIGNATED_REPORTING_DURATION_SECONDS()
+    market = localFixture.upload('../source/contracts/reporting/Market.sol', 'market')
+    market.setController(localFixture.contracts["Controller"].address)
+    with raises(TransactionFailed, message="outcomes has to be greater than 1"):
+        market.initialize(mockReportingWindow.address, endTime, 1, numTicks, fee, mockCash.address, tester.a1, tester.a1)
+
+    with raises(TransactionFailed, message="outcomes has to be less than 9"):
+        market.initialize(mockReportingWindow.address, endTime, 9, numTicks, fee, mockCash.address, tester.a1, tester.a1)
+
+    with raises(TransactionFailed, message="numTicks needs to be divisable by outcomes"):
+        market.initialize(mockReportingWindow.address, endTime, 7, numTicks, fee, mockCash.address, tester.a1, tester.a1)
+
+    with raises(TransactionFailed, message="fee per eth can not be greater than max fee per eth in attoEth"):
+        market.initialize(mockReportingWindow.address, endTime, 5, numTicks, 5, mockCash.address, tester.a1, tester.a1)
+
+    with raises(TransactionFailed, message="creator address can not be 0"):
+        market.initialize(mockReportingWindow.address, endTime, 5, numTicks, 5, mockCash.address, longToHexString(0), tester.a1)
+
+    with raises(TransactionFailed, message="designated reporter address can not be 0"):
+        market.initialize(mockReportingWindow.address, endTime, 5, numTicks, 5, mockCash.address, tester.a1, longToHexString(0))
+
+    mockUniverse.setForkingMarket(mockMarket.address)
+    mockReportingWindow.setUniverse(mockUniverse.address)
+    with raises(TransactionFailed, message="forking market address has to be 0"):
+        market.initialize(mockReportingWindow.address, endTime, 5, numTicks, 5, mockCash.address, tester.a1, tester.a1)
+
+    mockUniverse.setForkingMarket(longToHexString(0))
+    mockReputationToken.setBalanceOf(0)
+    mockUniverse.setDesignatedReportNoShowBond(100)
+    mockReportingWindow.setReputationToken(mockReputationToken.address)
+    with raises(TransactionFailed, message="reporting window reputation token does not have enough balance"):
+        market.initialize(mockReportingWindow.address, endTime, 5, numTicks, 5, mockCash.address, tester.a1, tester.a1)
+    
+    mockReputationToken.setBalanceOf(100)
+    mockUniverse.setTargetReporterGasCosts(15)
+    mockUniverse.setValidityBond(12)
+    with raises(TransactionFailed, message="refund is not over 0"):
```

Comment:
over or equal to 0 just to be nit picky. Really the throw is because they didn't provide sufficient ETH to pay the bond costs